src/models/chainspec.rs

```diff
@@ -0,0 +1,470 @@
+    eips: HashMap<String, String>,
+    max_code_size: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct Params {
+    account_start_nonce: String,
+    chain_id: String,
+    gas_limit_bound_divisor: String,
+    max_code_size: String,
+    maximum_extra_data_size: String,
+    min_gas_limit: String,
+    network_id: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct Genesis {
+    gas_limit: String,
+    timestamp: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct Precompiles {
+    name: String,
+    pricing: Pricing,
+}
+
+#[derive(Debug, Deserialize)]
+struct Pricing {
+    formula: String,
+    params: PricingParams,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+#[serde(untagged)]
+enum PricingParams {
+    ModExp { divisor: f32 },
+    Linear { base: f32, word: f32 },
+    Price { price: f32 },
+    BasePair { base: f32, pair: f32 },
+    GasPerRound { gas_per_round: f32 },
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn load_chainspec_struct() {
+        let chain_spec_str = r#"
+        name = "Rinkeby"
+        data_dir = "rinkeby"
+        
+        bootnodes = [
+          "enode://a24ac7c5484ef4ed0c5eb2d36620ba4e4aa13b8c84684e1b4aab0cebea2ae45cb4d375b77eab56516d34bfbd3c1a833fc51296ff084b770b94fb9028c4d25ccf@52.169.42.101:30303",
+          "enode://343149e4feefa15d882d9fe4ac7d88f885bd05ebb735e547f12e12080a9fa07c8014ca6fd7f373123488102fe5e34111f8509cf0b7de3f5b44339c9f25e87cb8@52.3.158.184:30303",
+          "enode://b6b28890b006743680c52e64e0d16db57f28124885595fa03a562be1d2bf0f3a1da297d56b13da25fb992888fd556d4c1a27b1f39d531bde7de1921c90061cc6@159.89.28.211:30303",
+        ]
+        
+        [engine]
+        name = "clique"
+        [engine.params]
+        period = 15.0
+        epoch = 30000.0
+        [engine.params.genesis]
+        vanity = "0x52657370656374206d7920617574686f7269746168207e452e436172746d616e"
+        signers = [
+            "0x42eb768f2244c8811c63729a21a3569731535f06",
+            "0x7ffc57839b00206d1ad20c69a1981b489f772031",
+            "0xb279182d99e65703f0076e4812653aab85fca0f0",
+        ]
+        
+        [hardforks]
+        eip140 = "0xfcc25"
+        eip145 = "0x37db77"
+        eip150 = "0x2"
+        eip155 = "0x3"
+        eip160 = "0x0"
+        eip161abc = "0x0"
+        eip161d = "0x0"
+        eip211 = "0xfcc25"
+        eip214 = "0xfcc25"
+        eip658 = "0xfcc25"
+        eip1014 = "0x37db77"
+        eip1052 = "0x37db77"
+        eip1283 = "0x37db77"
+        eip1283_disable = "0x41efd2"
+        eip1283_reenable = "0x52efd1"
+        eip1344 = "0x52efd1"
+        eip1706 = "0x52efd1"
+        eip1884 = "0x52efd1"
+        eip2028 = "0x52efd1"
+        max_code_size = "0x0"
+        
+        [params]
+        account_start_nonce = "0x0"
+        chain_id = "0x4"
+        gas_limit_bound_divisor = "0x400"
+        max_code_size = "0x6000"
+        maximum_extra_data_size = "0xffff"
+        min_gas_limit = "0x1388"
+        network_id = "0x4"
```

Comment:
@vorot93 can we have something a bit more specific than a string for hex encoded decimals?

Reply:
Yeah, it should be `u64`, or `primitive_types::U64` since it has relevant decoding code.

Reply:
How do you think, using u64 or primitive_types::U64 will make this ambiguous? Both u64 and string representation will be possible.